api/v1alpha1/zz_generated.deepcopy.go

```diff
@@ -127,6 +127,12 @@ func (in *DatasourceList) DeepCopyObject() runtime.Object {
 // DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
 func (in *DatasourceSpec) DeepCopyInto(out *DatasourceSpec) {
 	*out = *in
+	in.Enpoint.DeepCopyInto(&out.Enpoint)
+	if in.OSS != nil {
+		in, out := &in.OSS, &out.OSS
+		*out = new(OSS)
+		**out = **in
+	}
```

Comment:
**Suggestion:** The `DeepCopyInto` method for `DatasourceSpec`, `EmbedderSpec`, `LLMSpec` and `OSS` are not handling the case where the receiver (`in`) is `nil`. This could lead to a nil pointer dereference panic. It's better to add a nil check at the beginning of these methods.
```suggestion
func (in *DatasourceSpec) DeepCopyInto(out *DatasourceSpec) {
  if in == nil {
    return
  }
  *out = *in
  in.Enpoint.DeepCopyInto(&out.Enpoint)
  if in.OSS != nil {
    in, out := &in.OSS, &out.OSS
    *out = new(OSS)
    **out = **in
  }
}
```